addon/model.js

```diff
@@ -467,25 +587,165 @@ export default class MegamorphicModel extends EmberObject {
+    );
+  });
+} else {
+  isDirty = retrieveFromCurrentState;
+}
+
+let isDeleted;
+if (false) {
+  isDeleted = computed(function() {
+    return this._recordData.isDeleted();
+  });
+} else {
+  isDeleted = retrieveFromCurrentState;
+}
+
+let isNew;
+if (false) {
+  isNew = computed(function() {
+    return this._recordData.isNew();
+  });
+} else {
+  isNew = retrieveFromCurrentState;
+}
+
+let isSaving;
+if (false) {
+  isSaving = computed(function() {
+    let requests = this.store
+      .getRequestStateService()
+      .getPendingRequestsForRecord(this._identifier);
+    return !!requests.find(req => req.request.data[0].op === 'saveRecord');
+  });
+} else {
+  isSaving = retrieveFromCurrentState;
+}
+
+let isLoaded;
+if (false) {
+  isLoaded = computed(function() {
+    return this._recordData._pushed;
+  });
+} else {
+  isLoaded = retrieveFromCurrentState;
+}
+
+let isLoading;
+if (false) {
+  isLoading = computed(function() {
+    return !this.get('isLoaded');
+  });
+} else {
+  isLoading = retrieveFromCurrentState;
+}
+
+let dirtyType;
+if (false) {
+  dirtyType = computed(function() {
+    if (this._recordData.isNew()) {
+      return 'created';
+    }
+    if (this._recordData.isDeleted()) {
+      return 'deleted';
+    }
+    if (this._recordData.hasChangedAttributes()) {
+      return 'updated';
+    }
+  }));
+} else {
+  dirtyType = retrieveFromCurrentState;
+}
+
+let currentState;
+if (false) {
+  currentState = computed(function() {
+    let stateName = 'root';
+    stateName = stateName + '.loaded';
+    if (this._recordData.hasChangedAttributes()) {
+      stateName = stateName + '.updated.uncommitted';
+    } else {
+      stateName = stateName + '.saved';
+    }
+    return {
+      stateName,
+    };
+  });
+} else {
+  currentState = retrieveFromCurrentState;
+}
 // STATE PROPS
-defineProperty(MegamorphicModel.prototype, 'isEmpty', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isLoading', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isLoaded', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isSaving', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isDeleted', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isNew', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isValid', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'isDirty', retrieveFromCurrentState);
-defineProperty(MegamorphicModel.prototype, 'dirtyType', retrieveFromCurrentState);
+defineProperty(MegamorphicModel.prototype, 'isLoading', isLoaded);
+defineProperty(MegamorphicModel.prototype, 'isLoaded', isLoading);
+defineProperty(MegamorphicModel.prototype, 'dirtyType', dirtyType);
+
+defineProperty(MegamorphicModel.prototype, 'isDirty', isDirty);
+defineProperty(
+  MegamorphicModel.prototype,
+  'isEmpty',
+  computed(function() {
+    return false;
+  })
```

Comment:
this probably doesn't need to be a computed property

Reply:
fixed